pyrene/src/components/DateTimePicker/DateTimeInput/DateTimeInput.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { forwardRef } from 'react';
+import clsx from 'clsx';
+
+import Icon from '../../Icon/Icon';
+import styles from './dateTimeInput.css';
+
+const allowedValueCheck = (valueToCheck:string) : boolean => (/^[0-9.: APM]*$/.test(valueToCheck));
+
+export interface DateTimeInputProps {
+  /**
+   * Boolean to toggle time display
+   */
+  dateOnly?: boolean,
+  dateValue?: string,
+  errorValue?: string,
+  handleOn?: (dateString: string, timeString: string) => void
+  disabled?: boolean,
+  label?: string,
+  name?: string,
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void, // Handle change function passed from react-datepicker
+  onClick?: () => void,
+  onFocus?: () => void,
+  range?: boolean,
+  setDateValue?: (value: string) => void,
+  setTimeValue?: (value: string) => void,
+  tabNum?: number,
+  timeValue?: string,
+  value?: string
+}
+
+const DateTimeInput = forwardRef(({
+  dateOnly = false,
+  dateValue = '',
+  errorValue = '',
+  handleOn,
+  disabled,
+  label,
+  name = '',
+  onChange = () => {},
+  onClick = () => {},
+  onFocus,
+  setDateValue = () => {},
+  setTimeValue = () => {},
+  tabNum,
+  timeValue = '',
+}:DateTimeInputProps, ref:React.Ref<HTMLInputElement>) => {
+
+  const handleDateOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const node = event.target as HTMLInputElement;
+
+    if (allowedValueCheck(node.value)) {
+      if (!dateOnly && node.value.length > 10) {
+        setDateValue(node.value.substring(0, 10).trim());
+        setTimeValue(node.value.substring(10));
+
+        if (node.value.substring(10).trim().length >= 5) {
+          return onChange(event);
+        }
+      } else {
+        setDateValue(node.value);
+        setTimeValue('');
+
+        return onChange(event);
+      }
+    }
+    return null;
+  };
+
+  const formatTime = (value: string) => {
```

Comment:
I do not get what the `formatTime` does.

Reply:
This function is more for aesthetics. How it works is this: if the user forgets to put a space between the date and time it automatically inserts it. Example, if user types '11.01.200111:30' it will be separated to '11.01.2001 11:30' this is not important and can be removed.

Reply:
✅